addons/account/models/account_move.py

```diff
@@ -2105,6 +2111,29 @@ def _validate_taxes_country(self):
     # -------------------------------------------------------------------------
     # LOW-LEVEL METHODS
     # -------------------------------------------------------------------------
+    def _get_debit_from_balance(self, balance):
+        compare_res = self.env.company.currency_id.compare_amounts(balance, 0.0)
```

Comment:
The company of the move created/updated is not always the one of the company. This can cause issues in multi company environments

Reply:
indeed, must use self.company_id

Reply:
> indeed, must use self.company_id

It doesn't work if you are creating (so `self` is an empty recordset) a record for a company that is not the current main company in the env. (so you can't get it from `self.env.company` either)